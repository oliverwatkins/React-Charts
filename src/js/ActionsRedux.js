"use strict";

export const MiscActions = {
  FETCH_BAR_DATA : 'FETCH_BAR_DATA',
  SERVER_RESPONSE : 'SERVER_RESPONSE',
}

//pie
export const PieActions = {
  ENTER_PIE_CHART_NAME: 'ENTER_PIE_CHART_NAME',
  ADD_PIE_SLICE: 'ADD_PIE_SLICE',
  CHANGE_NAME_PIE: 'CHANGE_NAME_PIE',
  CREATE_SLICE: 'CREATE_SLICE',
  DELETE_SLICE: 'DELETE_SLICE',
  UPDATE_COLOR: 'UPDATE_COLOR'
}

//bar
export const BarActions = {
  CHANGE_NAME_BAR: 'CHANGE_NAME_BAR',
  CREATE_SERIES: 'CREATE_SERIES',
  CELL_CHANGED: 'CELL_CHANGED',
  CREATE_CATEGORY: 'CREATE_CATEGORY',
  DELETE_SERIES: 'DELETE_SERIES',
  DELETE_CATEGORY: 'DELETE_CATEGORY',
  UPDATE_COLOR: 'UPDATE_COLOR'
}


//pie
export function enterPieChartName() {
  return {type: PieActions.ENTER_PIE_CHART_NAME}
}

export function addSlice(id) {
  return {
    type: PieActions.ADD_PIE_SLICE,
    id
  }
}
export function createSlice(slice) {
  return {
    type: PieActions.CREATE_SLICE,
    slice
  }
}
export function deleteSlice(id) {
  return {
    type: PieActions.CREATE_SLICE,
    id
  }
}
export function changeNamePie(newName) {
  return {
    type: PieActions.CHANGE_NAME_PIE,
    newName
  }
}


//bar
export function createSeries(series) {
  return {
    type: BarActions.CREATE_SERIES,
    series
  }
}

export function deleteSeries(seriesName) {
  return {
    type: BarActions.DELETE_SERIES,
    seriesName
  }
}

export function updateColorBar(color, seriesName) {
  return {
    type: BarActions.UPDATE_COLOR,
    seriesName,
    color
  }
}

export function updateColorPie(color, seriesName) {
  return {
    type: PieActions.UPDATE_COLOR,
    seriesName,
    color
  }
}

export function changeLineChartName(newName) {
  return {
    type: BarActions.CHANGE_NAME_BAR,
    newName
  }
}

export function createCategory(value) {
  return {
    type: BarActions.CREATE_CATEGORY,
    value
  }
}

export function deleteCategory(categoryName, index) {
  return {
    type: BarActions.DELETE_CATEGORY,
    categoryName,
    index
  }
}


export function changeCell(value) {
  return {
    type: BarActions.CELL_CHANGED,
    value
  }
}
export function fetchBarData(value) {
  return {
    type: MiscActions.FETCH_BAR_DATA,
    value
  }
}

