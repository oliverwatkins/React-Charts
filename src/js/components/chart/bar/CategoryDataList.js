import React from "react";

import BarChartEntity from "../../../../js/entity/BarChartEntity";

import './List.less';
import Actions from "../../../../js/Actions";

export default class CategoryDataList extends React.Component {
  constructor(props) {
    super(props);
  }

  componentWillMount() {
    // SlicesStore.on("change", this.getSlices);
  }

  componentWillUnmount() {
    // SlicesStore.removeListener("change", this.getSlices);
  }


  deleteCategory(event, categoryName, index) {
    event.preventDefault();
    Actions.deleteCategory(categoryName, index);
  }

  render() {
    // const series = BarChartEntity.getSeries(this.props.app);
    const categories = BarChartEntity.getCategories(this.props.app);

    let deleteS = this.deleteCategory;

    return (
    <div className="listStyle">
      <table>
        <tbody>
        {categories.map(function (cat, i) {
          let key = 'xx_' + i;
          return (
            <tr key={key}>
              <td>
                {cat}
              </td>
              <td>
                <input type="button" value="delete"
                       onClick={(e) => deleteS(e, cat, i)}/>
              </td>
            </tr>
          );
        })}

        </tbody>
      </table>
    </div>
    );
  }
}
