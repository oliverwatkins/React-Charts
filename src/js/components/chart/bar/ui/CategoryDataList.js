import React from "react";

import './BarSeriesList.less';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import FlatButton from 'material-ui/FlatButton';
import PropTypes from 'prop-types';

export class CategoryDataList extends React.Component {
  constructor(props) {
    super(props);
  }

  static propTypes = {
    categories: PropTypes.array.isRequired,
    deleteCategory: PropTypes.func
  };

  render() {
    const categories = this.props.categories;

    let deleteCat = this.props.deleteCategory;
    return (
      <MuiThemeProvider>
        <div className="listStyle">
          <table>
            <tbody>
            {categories.map(function (cat, i) {
              let key = 'key_' + i;
              return (
                <tr key={key}>
                  <td>
                    {cat}
                  </td>
                  <td>
                    <FlatButton label="Delete" secondary={true}
                                onClick={(e) => deleteCat(cat, e, i)}/>

                    {/*<input type="button" value="delete"*/}
                    {/*onClick={(e) => deleteS(e, cat, i)}/>*/}
                  </td>
                </tr>
              );
            })}

            </tbody>
          </table>
        </div>
      </MuiThemeProvider>

    );
  }

  deleteCategory(event, categoryName, index) {
    event.preventDefault();
    this.props.deleteCategory(categoryName, index);
  }
}

export default CategoryDataList;






