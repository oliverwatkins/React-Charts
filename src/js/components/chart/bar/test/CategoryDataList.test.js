import React from 'react';
import {CategoryDataList} from './../CategoryDataList'; //is this dumb?
import renderer from 'react-test-renderer';


import * as actions from '../../../../ActionsRedux'




test('LinkXX changes the class when hovered', () => {
  const component = renderer.create(
    <CategoryDataList>Facebook</CategoryDataList>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});