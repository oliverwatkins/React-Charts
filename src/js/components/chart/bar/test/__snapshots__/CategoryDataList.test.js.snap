// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test basic rendering CategroyDataList renders 1`] = `
<div
  className="listStyle"
>
  <table>
    <tbody>
      <tr>
        <td>
          Apple
        </td>
        <td>
          <button
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "backgroundColor": "rgba(0, 0, 0, 0)",
                "border": 10,
                "borderRadius": 2,
                "boxSizing": "border-box",
                "color": "#ff4081",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": "inherit",
                "fontWeight": "inherit",
                "height": 36,
                "lineHeight": "36px",
                "margin": 0,
                "minWidth": 88,
                "muiPrepared": true,
                "outline": "none",
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "textAlign": "center",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "verticalAlign": null,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <span
                style={
                  Object {
                    "fontSize": 14,
                    "fontWeight": 500,
                    "letterSpacing": 0,
                    "muiPrepared": true,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "position": "relative",
                    "textTransform": "uppercase",
                    "verticalAlign": "middle",
                  }
                }
              >
                Delete
              </span>
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Orange
        </td>
        <td>
          <button
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "backgroundColor": "rgba(0, 0, 0, 0)",
                "border": 10,
                "borderRadius": 2,
                "boxSizing": "border-box",
                "color": "#ff4081",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": "inherit",
                "fontWeight": "inherit",
                "height": 36,
                "lineHeight": "36px",
                "margin": 0,
                "minWidth": 88,
                "muiPrepared": true,
                "outline": "none",
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "textAlign": "center",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "verticalAlign": null,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <span
                style={
                  Object {
                    "fontSize": 14,
                    "fontWeight": 500,
                    "letterSpacing": 0,
                    "muiPrepared": true,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "position": "relative",
                    "textTransform": "uppercase",
                    "verticalAlign": "middle",
                  }
                }
              >
                Delete
              </span>
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          tomato
        </td>
        <td>
          <button
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "backgroundColor": "rgba(0, 0, 0, 0)",
                "border": 10,
                "borderRadius": 2,
                "boxSizing": "border-box",
                "color": "#ff4081",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": "inherit",
                "fontWeight": "inherit",
                "height": 36,
                "lineHeight": "36px",
                "margin": 0,
                "minWidth": 88,
                "muiPrepared": true,
                "outline": "none",
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "textAlign": "center",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "verticalAlign": null,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <span
                style={
                  Object {
                    "fontSize": 14,
                    "fontWeight": 500,
                    "letterSpacing": 0,
                    "muiPrepared": true,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "position": "relative",
                    "textTransform": "uppercase",
                    "verticalAlign": "middle",
                  }
                }
              >
                Delete
              </span>
            </div>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
